File: Oidc.AuthorizationServer/Models/IdentityProviderRestriction.cs
using FluentNHibernate.Data;
using lvl.Ontology;
using lvl.Ontology.Authorization;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace lvl.Oidc.AuthorizationServer.Models
{
    /// <summary>
    ///     Represents an independent service provider which is allowed.
    /// </summary>
    [Table("IdentityProvider", Schema = "oidc")]
    [HiddenFromApi]
    public class IdentityProviderRestriction : Entity, IAggregateScope<ClientEntity>
    {
        /// <summary>
        ///     The name of an independeant service provider.
        /// </summary>
        [Required]
        public string Name { get; set; }
    }
}

File: oidc/src/models/identity-provider-restriction.ts
import { ClientEntity } from './client-entity';
import { Entity, IAggregateScope, Required } from '@lvl/front-end';

export class IdentityProviderRestriction extends Entity implements IAggregateScope<ClientEntity> {
    @Required() public name: string;

    constructor(options?: IIdentityProviderRestrictionOptions) {
        options = options || {};
        super(options);
        this.name = options!.name;
    }
}

interface IIdentityProviderRestrictionOptions {
    name?: string;
    id?: number;
}
